### DISCUSSION TITLE
Django select2 field in pure html in modal with already used field_id

### DISCUSSION THREAD

**Discussion by Mergpijp** [codingjoe/django-select2#38] [Q&A]
**Goal**
I am trying to make select2 work in pure html in a modal.

**Problem**
I have a workaround by using another field_id from another form. Is this the right way? Do I need to create a field_Id for each select2 field?

**Code Snippet**
Please provide a code snippet of your problem:

                <div class="form-group" >
                   <label for="id_filecategory" class="">
                   Filecategory
                   </label>
                   <div class="">
                      <div class="input-group">
                         <select name="filecategory" data-minimum-input-length="0" data-allow-clear="true" data-placeholder="---------" data-ajax--url="/select2/fields/auto.json?field_id=Mjc5NDMxNzU1OTM0NA%3A1lEEm5%3AK_HUKQUF1aEwQZZGHySFs6tbuzc" data-ajax--cache="true" data-ajax--type="GET" class="modelselect2widget form-control django-select2 django-select2-heavy" id="id_filecategory">
                            <option value=" " selected></option>
                         </select>
                         <span class="input-group-append"><button  class="btn btn-danger" onClick="window.open(&#39;/filecategory/new&#39;, &#39;_blank&#39;, &#39;width=1000,height=600,menubar=no,toolbar=no&#39;);" type="button">+</button>
                         </span>
                      </div>
                   </div>
                </div>

**Comment by codingjoe** [comment]
Hi @Mergpijp,

Thanks for reaching out. I am not 100% sure that I fully understood your questions. You want to use django-select2 without a Django form in the backend? If so, you will need to provide your own JSON view to provide the data for the field. You find an example on how this is done in the docs for the `HeavySelectMixin`, see also 
https://github.com/codingjoe/django-select2/blob/e735be2016f0a2349f0f8acd51eb6b51741cc123/django_select2/forms.py#L283-L301

I hope that helps you solve your problem.

Best,
Joe

**Reply by Mergpijp** [reply]
Thanks for reaching out but I do not get it. Previously I had a crispy form with his:
`    filecategory = forms.ModelChoiceField(widget=ModelSelect2Widget(
        model=FileCategory,
        search_fields=['name__icontains', ],
        attrs={'data-minimum-input-length': 0},
    ), queryset=FileCategory.objects.all(), required=False)`

How do I provide my own Json view?

**Reply by Mergpijp** [reply]
The question is what do I fill in in 'my_view_name' or data url. how do I configure my html in such way that it communicates with the python code you provided.

**Reply by Mergpijp** [reply]
Also I could not find a example other then you provided in the docs.

**Reply by Mergpijp** [reply]
Okay I found something of a colleague of mine. [HeavySelect2MultipleWidget](https://github.com/Mediate18/Mediate/wiki/How-I-use-HeavySelect2MultipleWidget ) Now i need the view to communicate to the javascript and make my own html for the select field.

**Reply by Mergpijp** [reply]
So I solved this by using the javascript select2 ajax code. I only made a django view that gets all filecategory and returns a jsonresponse. I did nothing with forms  Thanks for your help anyway.
